### PR TITLE
Fix recent coverity issues and couple of assorted changes in p11test

### PR DESCRIPTION
**Pull request by Jakuje**
including some of minor reformatting and trailing whitespace removal in separate commits.

##### Checklist
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: src/smm/sm-card-authentic.c [hunk 1]
```diff
@@ -80,19 +79,23 @@ sm_oberthur_diversify_keyset(struct sc_context *ctx, struct sm_info *sm_info,
 			sc_debug(ctx, SC_LOG_DEBUG_SM, "key_buf:%s", sc_dump_hex(key_buff, 16));
 
 			rv = sm_encrypt_des_ecb3(master_key, key_buff, sizeof(key_buff), &tmp, &tmp_len);
-			LOG_TEST_RET(ctx, rv, "GP init session: cannot derive key");
+			if (rv != 0) {
+				free(tmp);
```
[the remark this answers is not in the document]
Comment (Jakuje) [reply src/smm/sm-card-authentic.c:83]: Good point. I will fix that. Thanks for checking!

File: src/sm/sm-common.c [hunk 2]
```diff
@@ -385,29 +383,32 @@ sm_encrypt_des_ecb3(unsigned char *key, unsigned char *data, int data_len,
 	for (ii=0; ii<data_len; ii+=8)
 		DES_ecb2_encrypt( (sm_des_cblock *)(data + ii),
 				(sm_des_cblock *)(*out + ii), &ks, &ks2, DES_ENCRYPT);
+
+	return 0;
```
Comment (frankmorgner) [inline src/sm/sm-common.c:387]: better use SC_SUCCESS here

File: src/sm/sm-common.c [hunk 3]
```diff
@@ -385,29 +383,32 @@ sm_encrypt_des_ecb3(unsigned char *key, unsigned char *data, int data_len,
...
 #else
 	cctx = EVP_CIPHER_CTX_new();
 	if (!EVP_EncryptInit_ex2(cctx, EVP_des_ede_ecb(), key, NULL, NULL)) {
```
Comment (frankmorgner) [inline src/sm/sm-common.c:390]: please add a check for `NULL == cctx`

File: src/sm/sm-common.c [hunk 4]
```diff
@@ -385,29 +383,32 @@ sm_encrypt_des_ecb3(unsigned char *key, unsigned char *data, int data_len,
...
+		goto err;
 	}
 	*out_len += tmplen;
 	EVP_CIPHER_CTX_free(cctx);
-#endif
-
 	return 0;
+
+err:
+	EVP_CIPHER_CTX_free(cctx);
```
Comment (frankmorgner) [inline src/sm/sm-common.c:408]: please check with `if (cctx)`

**Review by frankmorgner — CHANGES_REQUESTED** [review]

File: src/sm/sm-common.c (hunk 4 above)
Comment (Jakuje) [reply src/sm/sm-common.c:411]: Unfortunately, the manual page does not specify this, but OpenSSL API is getting sane semantics and NULL arguments are ok for the free functions:
https://github.com/openssl/openssl/blob/f5b06306b7727f830407c639f8a7bf1698f8aaa5/crypto/evp/evp_enc.c#L84

File: src/sm/sm-common.c (hunk 2 above)
Comment (Jakuje) [reply src/sm/sm-common.c]: Thanks for review. Changed to use SC_SUCCESS also in other cases of this function.

File: src/sm/sm-common.c (hunk 4 above)
Comment (frankmorgner) [reply src/sm/sm-common.c:411]: Nice to know; is this a general approach for all APIs? Historically, they typically didn't check any input parameters...

Comment (Jakuje) [reply src/sm/sm-common.c:411]: I am not aware of specific OpenSSL guidelines, but I think it follows the least surprise principle as the `free()` ignores the `NULL` pointers and people are (finally) getting use to that too as it simplifies a lot of needless null checks on cleanup.

**Review by frankmorgner — APPROVED** [review]

**Comment by Jakuje** [comment]
Thank you for the review. Merging to move on with this.

**Pull request merged**